bundle/Form/DataMapper.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\InformationCollectionBundle\Form;
+
+use DateTime;
+use Symfony\Component\Form\DataMapperInterface;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\PropertyAccess\PropertyPathInterface;
+use function is_array;
+use function is_object;
+
+/**
+ * A data mapper using property paths to read/write data.
+ */
+abstract class DataMapper implements DataMapperInterface
+{
+    protected FieldTypeHandlerRegistry $fieldTypeHandlerRegistry;
+
+    protected PropertyAccessorInterface $propertyAccessor;
+
+    /**
+     * Creates a new property path mapper.
+     */
+    public function __construct(
+        FieldTypeHandlerRegistry $fieldTypeHandlerRegistry,
+        ?PropertyAccessorInterface $propertyAccessor = null
+    ) {
+        $this->fieldTypeHandlerRegistry = $fieldTypeHandlerRegistry;
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+    }
+
+    public function mapDataToForms($data, $forms): void
+    {
+        $empty = null === $data || [] === $data;
+
+        if (!$empty && !is_array($data) && !is_object($data)) {
+            throw new UnexpectedTypeException($data, 'object, array or empty');
+        }
+
+        foreach ($forms as $form) {
+            $propertyPath = $form->getPropertyPath();
+            $config = $form->getConfig();
+
+            if ($data instanceof DataWrapper && null !== $propertyPath && $config->getMapped()) {
+                /* @var $data \Netgen\Bundle\InformationCollectionBundle\Form\DataWrapper */
+                $this->mapToForm($form, $data, $propertyPath);
+            } elseif (!$empty && null !== $propertyPath && $config->getMapped()) {
+                $form->setData($this->propertyAccessor->getValue($data, $propertyPath));
+            } else {
+                $form->setData($form->getConfig()->getData());
+            }
+        }
+    }
+
+    public function mapFormsToData($forms, &$data): void
```

Comment:
Use signature from the interface here.

Reply:
Solved in d42fe134

Reply:
`@data` is `$viewData` in the interface.

Reply:
fixed in 6580c585